packages/headless/src/modules/customer/services/index.ts

```diff
@@ -44,8 +44,8 @@ export class CustomerService {
     }
 
     async addCustomerNewAddress(customerId: string, address: CustomerAddress): Promise<AddCustomerNewAddressResponse> {
-        let customer = await this.customerRepo.findCustomer({ id: customerId });
-        if (!customer) return this.helper.serviceResponse.errorResponse(GetCustomerInformationErrorMessages.CUSTOMER_NOT_FOUND, null, HttpStatus.BAD_REQUEST);
+        const doesDefaultAddressExists = address.isDefault && await this.customerRepo.findCustomer({ id: customerId, 'addresses.isDefault': true });
+        if (doesDefaultAddressExists) return this.helper.serviceResponse.errorResponse(AddCustomerNewAddressErrorMessages.ALREADY_DEFAULT_ADDRESS_EXISTS, null, HttpStatus.BAD_REQUEST);
```

Comment:
dont return error, rather make the new ones isDefault: true and the old ones isDefault: false